### ISSUE TITLE
#or issue with ^:ref

### ISSUE THREAD

**Issue by vincentjames501** [juxt/aero#23]
Piggybacking off #17 , I think it would make sense for the following to work:

config.edn

``` clojure
{:test-val nil
 :actual-val #or [^:ref [:test-val] "abc123"]}
```

Result:

``` clojure
(aero/read-config (io/resource "config.edn") {:profile  (keyword (env/env :mode))
                                                                       :resolver aero/resource-resolver})
=> {:test-val nil, :actual-val nil}
```

**Comment by vincentjames501** [comment]
#18 would also help alleviate what I'm trying to do.

**Comment by timgilbert** [comment]
I'm seeing a similar issue with `#join`. I've got something similar to:

``` clojure
{:root-path "https://rather-long-url-i-would-rather-not/type/"
 :uri #profile {:dev #join [^:ref [:root-path] "dev/my-special-path"]
                :qa  #join [^:ref [:root-path] "qa/my-special-path"]}}
```

But it looks like after the `#join` runs, the metadata is lost and the ref replacement never happens. I imagine this is due to the tag literals running before the ref replacement in `(get-in-ref)`? 

I'm not sure whether @weavejester's proposed solution for #17 would actually make a difference to this, at least not without adding a bunch of code to make all of the tag literals check their arguments for `(instance? Ref %)`.

**Comment by expez** [comment]
I also hit this with `#join` and was surprised it didn't work.

**Comment by SevereOverfl0w** [comment]
`#join` + `#ref` should be fixed with 1.1.0, can you please test? refs are now ordered.

**Comment by timgilbert** [comment]
I'll check it out on my end. Is there a changelog anywhere? The README still refers to using `^:ref`, I take it that should be `#ref` now?

**Comment by SevereOverfl0w** [comment]
Yep. I just haven't taken the time to get the latest release all nicked up yet.

On this project, the changelog is the git log pretty much. @malcolmsparks has been experimenting with keeping Changelogs on other projects though.

**Comment by vise890** [comment]
```clojure
{:foo #or [#env FOO 1]
 :bar #ref [:foo]}
```
that leaves me with 
```clojure
{:foo nil
 :bar nil}
```
am I missing something?

**Comment by SevereOverfl0w** [comment]
Regarding the original issue here:

```
(read-config
    (StringReader. "{:foo #or [#ref [:bar] :other] :bar nil}"))
```

Gives me:

```
{:foo :other, :bar nil}
```

So I think is resolved.

**Comment by expez** [comment]
> #join + #ref should be fixed with 1.1.0, can you please test? refs are now ordered.

Confirmed working.

Thanks! :+1:

**Comment by SevereOverfl0w** [comment]
Closing this as I expect it to be fixed.
